### PR TITLE
⭐️ extend okta policy to expose more information

### PR DESCRIPTION
**Pull request by chris-rock**

### CODE REVIEW

File: resources/packs/okta/sdk/policies.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/okta/okta-sdk-golang/v2/okta"
+	"github.com/okta/okta-sdk-golang/v2/okta/query"
+)
+
+// SkdExtension handles cases where Okta's SDK doesn't support a particular API
+type SkdExtension struct {
```
Comment (preslavgerchev) [inline resources/packs/okta/sdk/policies.go]: nit: 
```suggestion
type SdkExtension struct {
```

File: resources/packs/okta/sdk/policies.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+		return nil, err
+	}
+
+	var settingsJSON []byte
+	if a.Settings != nil {
+		type Alias PolicySettings
+		type local struct {
+			*Alias
+		}
+		result := local{Alias: (*Alias)(a.Settings)}
```
Comment (preslavgerchev) [inline resources/packs/okta/sdk/policies.go]: i struggle to understand this, looks like it's just a wrapper around an aliased policy settings.
wouldn't `result := a.Settings` output the same then?

Comment (chris-rock) [reply resources/packs/okta/sdk/policies.go]: I simplifies the code and extended testing

File: resources/packs/okta/sdk/policies.go (hunk 1 above)
Comment (chris-rock) [reply resources/packs/okta/sdk/policies.go]: Great catch @preslavgerchev

**Comment by chris-rock** [comment]
@preslavgerchev code updated and it includes your feedback

File: resources/packs/okta/sdk/policies_test.go [hunk 3]
```diff
@@ -47,4 +47,9 @@ func TestPoliciesParsing(t *testing.T) {
 	err := decoder.Decode(&policies)
 	require.NoError(t, err)
 	assert.NotNil(t, policies[0].Settings.Password)
+
+	jData, err := json.Marshal(policies)
+	require.NoError(t, err)
+	assert.NotNil(t, jData)
+	assert.Contains(t, string(jData), "\"autoUnlockMinutes\":30,")
```
Comment (preslavgerchev) [inline resources/packs/okta/sdk/policies_test.go:54]: nicee

**Review by preslavgerchev — APPROVED** [review]

**Pull request merged**